src/core/SamplePreviewPlayHandle.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * SamplePreviewPlayHandle.cpp
+ *
+ * Copyright (c) 2025 Sotonye Atemie <sakertooth@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "SamplePreviewPlayHandle.h"
+
+#include "AudioBusHandle.h"
+#include "AudioEngine.h"
+#include "Engine.h"
+#include "SampleFrame.h"
+#include "ThreadPool.h"
+
+namespace lmms {
+SamplePreviewPlayHandle::SamplePreviewPlayHandle(const std::filesystem::path& path)
+	: PlayHandle(Type::SamplePlayHandle)
+{
+#ifdef LMMS_BUILD_WIN32
+	m_sndfile = sf_wchar_open(path.c_str(), SFM_READ, &m_sfinfo);
+#else
+	m_sndfile = sf_open(path.c_str(), SFM_READ, &m_sfinfo);
+#endif
```

Comment:
I guess the [`sf_open_virtual` function](https://libsndfile.github.io/libsndfile/api.html#open_virtual) isn't needed for libsndfile to stream from a file?

Reply:
I think it's for allowing use of libsndfile's interface with other audio sources, but here I need libsndfile's functionality.